src/pybind/mgr/smb/handler.py

```diff
@@ -324,6 +326,7 @@ def __init__(
         path_resolver: Optional[PathResolver] = None,
         authorizer: Optional[AccessAuthorizer] = None,
         orch: Optional[OrchSubmitter] = None,
+        earmark_resolver: Optional[CephFsEarmarkResolver] = None,
```

Comment:
I'd like to see the handler use abstract protocols in the types it accepts. This allows for better composition through component objects and the required methods. Ask me for more info if this is too vague.

Reply:
Gotcha! Added `EarmarkResolver` under proto.py